src/praisonai-bot/praisonai_bot/gateway/server.py

```
@@ -4155,45 +4155,68 @@ def _apply_hooks_from_config(self, cfg: Dict[str, Any]) -> None:
+        default on any build failure so inbound delivery keeps working.
+        """
+        if self._hook_idem is not None:
+            return self._hook_idem
+        backend = "memory"
+        try:
+            idem_cfg = getattr(self.config, "hook_idempotency", None)
+            if isinstance(idem_cfg, dict):
+                backend = str(idem_cfg.get("store_backend", "memory"))
+            elif idem_cfg is not None:
+                backend = str(getattr(idem_cfg, "store_backend", "memory"))
+        except Exception:  # pragma: no cover - defensive config access
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)